src/streams/streamsActions.js

```diff
@@ -6,26 +6,27 @@ import { getAuth, getZulipFeatureLevel } from '../selectors';
 export const updateExistingStream = (
   id: number,
   initialValues: Stream,
-  newValues: {| name: string, description: string, isPrivate: boolean |},
+  newValues: {| name: string, description: string, invite_only: boolean |},
 ): ThunkAction<Promise<void>> => async (dispatch, getState) => {
   const state = getState();
 
   const maybeEncode = (value: string): string =>
-    // Adapt to a server API change that was accidentally incompatible:
-    //   https://github.com/zulip/zulip-mobile/pull/4748#issuecomment-852254404
-    //   https://github.com/zulip/zulip-mobile/issues/4747#issuecomment-946362729
-    // TODO(#4659): Ideally this belongs inside `api.updateStream`.
-    // TODO(server-4.0): Simplify this (if it hasn't already moved.)
+    // See implementation comments in api.updateStream.
     getZulipFeatureLevel(state) >= 64 ? value : JSON.stringify(value);
```

Comment:
This new comment seems not specific enough to find what it's referring to and be sure you've found the thing that was intended -- there are a lot of comments on `api.updateStream` now :slightly_smiling_face:

Perhaps mention it's specifically the comment on `description` and `new_name`?

Hmm, also the new comment seems backward. The issue is that *old* servers required the superfluous `JSON.stringify`; new servers require that the parameter be simply the desired string.